cmd/kaspawallet/libkaspawallet/bip32/bip32_test.go

```diff
@@ -115,52 +207,54 @@ func TestBIP32SpecVectors(t *testing.T) {
 			t.Fatalf("DecodeString: %+v", err)
 		}
 
-		masterKey, err := NewMaster(seed, BitcoinMainnetPrivate)
+		masterKey, err := NewMaster(seed, vector.version)
 		if err != nil {
 			t.Fatalf("NewMaster: %+v", err)
 		}
 
-		for j, path := range vector.pathes {
-			extPrv, err := masterKey.Path(path.path)
+		for j, path := range vector.paths {
+			extendedPrivateKey, err := masterKey.DeriveFromPath(path.path)
 			if err != nil {
 				t.Fatalf("Path: %+v", err)
 			}
 
-			if extPrv.String() != path.extPrv {
-				t.Fatalf("Test (%d, %d): expected extPrv %s but got %s", i, j, path.extPrv, extPrv.String())
+			if extendedPrivateKey.String() != path.extendedPrivateKey {
+				t.Fatalf("Test (%d, %d): expected extended private key %s but got %s", i, j, path.extendedPrivateKey, extendedPrivateKey.String())
 			}
 
-			decodedExtPrv, err := DeserializeExtendedPrivateKey(extPrv.String())
+			decodedExtendedPrivateKey, err := DeserializeExtendedPrivateKey(extendedPrivateKey.String())
 			if err != nil {
 				t.Fatalf("DeserializeExtendedPrivateKey: %+v", err)
 			}
 
-			if extPrv.String() != decodedExtPrv.String() {
-				t.Fatalf("Test (%d, %d): deserializing and serializing the ext prv didn't preserve the data", i, j)
+			if extendedPrivateKey.String() != decodedExtendedPrivateKey.String() {
+				t.Fatalf("Test (%d, %d): deserializing and serializing the extended private key didn't preserve the data", i, j)
 			}
 
-			extPub, err := extPrv.Public()
+			extendedPublicKey, err := extendedPrivateKey.Public()
 			if err != nil {
 				t.Fatalf("Public: %+v", err)
 			}
 
-			if extPub.String() != path.extPub {
-				t.Fatalf("Test (%d, %d): expected extPub %s but got %s", i, j, path.extPub, extPub.String())
+			if extendedPublicKey.String() != path.extendedPublicKey {
+				t.Fatalf("Test (%d, %d): expected extended public key %s but got %s", i, j, path.extendedPublicKey, extendedPublicKey.String())
 			}
 
-			decodedExtPub, err := DeserializeExtendedPrivateKey(extPub.String())
+			decodedExtendedPublicKey, err := DeserializeExtendedPrivateKey(extendedPublicKey.String())
 			if err != nil {
 				t.Fatalf("DeserializeExtendedPublicKey: %+v", err)
 			}
 
-			if extPub.String() != decodedExtPub.String() {
+			if extendedPublicKey.String() != decodedExtendedPublicKey.String() {
 				t.Fatalf("Test (%d, %d): deserializing and serializing the ext pub didn't preserve the data", i, j)
 			}
 		}
 	}
 }
 
-func TestExtendedPublicKey_Path(t *testing.T) {
+// TestExtendedKey_DeriveFromPath checks that path that derive from extended public key and extended
```

Comment:
Suspected typo: derive from extended public key and extended public key

I think you want to replace one "public" to "private"